frontend/src/ui/busy/busy.tsx

```diff
@@ -10,4 +10,28 @@ export const BusyOverlay = () => (
     <div className="busy-overlay">
         <Spinner/>
     </div>
-);
\ No newline at end of file
+);
+
+export const BusyRender = <T extends {}>(props: {promise: Promise<T>, children: (data: T) => Array<ReactElement> | ReactElement}) => {
```

Comment:
* a better return type: `ReactNode` instead of `Array<ReactElement> | ReactElement`
https://stackoverflow.com/questions/58123398/when-to-use-jsx-element-vs-reactnode-vs-reactelement

* `<T extends {}>` => `<T extends unknown>` - has no assumptions on the type T. Yours assumes is object.
This is the prefered way of templating arrow functions in ts.